### PR TITLE
feat: add Banner component (#89)

### LINKED ISSUE #89: Create new banner component
Create a new banner component and storybook in the common component directory `packages/data-explorer-ui/src/components/common` to match mocks [here](https://www.figma.com/file/JOUDXkKfgivRNFdBsWDPEH/AnVIL-Data-Explorer?node-id=1-4432&t=U9YsfLdZ9NDzm3Fa-4) or [here](https://www.figma.com/file/JOUDXkKfgivRNFdBsWDPEH/AnVIL-Data-Explorer?node-id=1-257&t=U9YsfLdZ9NDzm3Fa-4).

### Approach
- Depends on #90 .

#### Part A
- Create a new component `Banner` in a new directory under `packages/data-explorer-ui/src/components/common`.
  - Add BannerProps interface, comprising of the following:
    - children - ReactNode, required.
- Create a styled component to add styles to the banner.
  - Note, the background colour should be `primary` from the theme (see `packages/data-explorer-ui/src/components/common/StatusIcon/statusIcon.styles.ts` for an example of theme usage).

#### Part B
- Create a new Button style `TextButtonWhite` in `packages/data-explorer-ui/src/components/common/Button/button.styles.ts` to match the call to action text link [here](https://www.figma.com/file/JOUDXkKfgivRNFdBsWDPEH/AnVIL-Data-Explorer?node-id=1-4432&t=U9YsfLdZ9NDzm3Fa-4).
- Add the new Button to SB `packages/data-explorer-ui/src/components/common/Button/button.stories.tsx`.

#### Part C
- Create a storybook for the new component.
  - The story should render the text as per the [mock](https://www.figma.com/file/JOUDXkKfgivRNFdBsWDPEH/AnVIL-Data-Explorer?node-id=1-4432&t=U9YsfLdZ9NDzm3Fa-4).
  - Use the CallToActionButton story (see #90) and the new button `TextButtonWhite` to render the message.

### Definition of Done
- Banner component matches mocks [here](https://www.figma.com/file/JOUDXkKfgivRNFdBsWDPEH/AnVIL-Data-Explorer?node-id=1-4432&t=U9YsfLdZ9NDzm3Fa-4) or [here](https://www.figma.com/file/JOUDXkKfgivRNFdBsWDPEH/AnVIL-Data-Explorer?node-id=1-257&t=U9YsfLdZ9NDzm3Fa-4).
- Banner SB matches mocks e.g. [here](https://www.figma.com/file/JOUDXkKfgivRNFdBsWDPEH/AnVIL-Data-Explorer?node-id=1-4432&t=U9YsfLdZ9NDzm3Fa-4).

### PR DESCRIPTION
**Pull request by imadx**
Closes #89 

Adds Banner component with `message` and `callToAction` props.

### CODE REVIEW

File: packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import styled from "@emotion/styled";
+import { Banner } from "./banner";
+
+export const BannerPrimary = styled(Banner)`
+  background-color: ${({ theme }) => theme.palette.primary.main};
+  color: ${({ theme }) => theme.palette.common.white};
+  padding: 4px 8px;
+  text-align: center;
+
+  & > a {
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:10]: Let's make the Banner children prop completely configurable - now that we have `TextButtonWhite` we no longer need to specify styles for an anchor. I suggest removing lines 10-15.

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { CallToActionButton } from "../Button/components/CallToActionButton/callToActionButton";
+import { Banner } from "./banner";
+import { BannerPrimary } from "./banner.styles";
+
+export default {
+  argTypes: {
+    children: { control: { disabled: true } },
+  },
+  component: Banner,
+  title: "Components/Common/Banner",
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:13]: Let's give the banner the opportunity to take full width of the viewport. Add the following to the story defaults.

```
  parameters: {
    layout: "fullscreen",
  },
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+} as ComponentMeta<typeof Banner>;
+
+const BannerPrimaryTemplate: ComponentStory<typeof Banner> = (args) => (
+  <Box sx={{ minWidth: 300 }}>
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:17]: We can now remove the Box around the template (because we would like the banner to use the full width of the viewport.

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+);
+
+export const BannerPrimaryStory = BannerPrimaryTemplate.bind({});
+BannerPrimaryStory.args = {
+  children: (
+    <>
+      <Typography component="span" variant="text-body-small-500">
+        Optional announcements text goes here followed by a
+      </Typography>{" "}
+      <CallToActionButton
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:29]: Let's use the CallToActionButton story here instead, and for the prop `ButtonElType` use `TextButtonWhite`.

**Review by frano-m — CHANGES_REQUESTED** [review]
Hi @imadx thank you so much! I have some minor feedback and then this will be ready to go!

File: packages/data-explorer-ui/src/components/common/Banner/banner.tsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Box } from "@mui/material";
+import React, { ReactNode } from "react";
+
+export interface BannerProps {
+  children: ReactNode;
+}
+
+export const Banner = ({ children, ...props }: BannerProps): JSX.Element => {
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.tsx:8]: Let's remove the prop spreading on this component - we can add it later if we need it. Thank you!

File: packages/data-explorer-ui/src/components/common/Button/button.stories.tsx [hunk 6]
```diff
@@ -23,6 +28,12 @@ const SecondaryButtonTemplate: ComponentStory<typeof Button> = (args) => (
   <ButtonSecondary {...args} />
 );
 
+const TextButtonWhiteTemplate: ComponentStory<typeof Button> = (args) => (
+  <Box sx={{ backgroundColor: "#333", borderRadius: 1, px: 10, py: 5 }}>
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/button.stories.tsx:32]: Let's remove the styles around the button template - just in case we want to use WhiteTextButtonStory elsewhere in another story.

File: packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import styled from "@emotion/styled";
+import { Banner } from "./banner";
+
+export const BannerPrimary = styled(Banner)`
+  background-color: ${({ theme }) => theme.palette.primary.main};
+  color: ${({ theme }) => theme.palette.common.white};
+  padding: 4px 8px;
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:7]: Let's make the padding 8px all around (I see Figma doesn't specify paddingX but let's make each side 8px).

Comment (imadx) [reply packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:7]: Hey @frano-m, I made this change. The height updated to `40px` with this however. The Figma has it set to `32px`. Shouldn't be an issue?

File: packages/data-explorer-ui/src/components/Layout/components/Header/components/ProfileComponent/components/AuthenticationMenu/authenticationMenu.stories.tsx [hunk 8]
```diff
@@ -15,7 +15,7 @@ export default {
   title: "Components/Navigation/AuthenticationMenu",
 } satisfies Meta<typeof AuthenticationMenu>;
 
-const AuthenticationMenuTemplate: StoryFn<typeof AuthenticationMenu> = (
+const AuthenticationMenuTemplate: ComponentStory<typeof AuthenticationMenu> = (
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/Layout/components/Header/components/ProfileComponent/components/AuthenticationMenu/authenticationMenu.stories.tsx:18]: Hi @imadx I think let's revert this - and keep the change requests focused to the ticket - we can resolve this in a new ticket if needed? Thoughts?

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx (hunk 4 above)
Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:30]: @imadx if we import the newly created TextButtonWhiteStory and replace ButtonElType={"TextButtonWhite"} with ButtonElType={TextButtonWhite}, I think this means we do not need to update or make changes to `CallToActionButton` component.

Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:30]: Let's also use the `CallToActionButtonStory` instead of `CallToActionButton` component on line `30`.

File: packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.tsx
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.tsx:1]: Hey @imadx I think we should revert the changes made to the `CallToActionButton` component - I'm not sure they are needed?

File: packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/calltoActionButton.stories.tsx
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/calltoActionButton.stories.tsx:1]: We probably don't need to create another CallToActionButton story - if we use `CallToActionButtonStory` we can override the arguments for that story. 😄

**Comment by NoopDog** [comment]
HI, @imadx looks like this PR needs a rebase! 
Cheers,
D

File: packages/data-explorer-ui/src/components/common/Banner/banner.tsx (hunk 5 above)
Comment (imadx) [reply packages/data-explorer-ui/src/components/common/Banner/banner.tsx]: The need for this was removed. I used `styled.div` here instead. Was much clean and simpler. Thanks!

File: packages/data-explorer-ui/src/components/common/Button/button.stories.tsx (hunk 6 above)
Comment (imadx) [reply packages/data-explorer-ui/src/components/common/Button/button.stories.tsx]: Cleaned this up

File: packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts (hunk 7 above)
Comment (imadx) [reply packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts]: Updated this. A `line-height` was added.

File: packages/data-explorer-ui/src/components/Layout/components/Header/components/ProfileComponent/components/AuthenticationMenu/authenticationMenu.stories.tsx (hunk 8 above)
Comment (imadx) [reply packages/data-explorer-ui/src/components/Layout/components/Header/components/ProfileComponent/components/AuthenticationMenu/authenticationMenu.stories.tsx]: Reverted this. Not required with the rebased updates of main

**Comment by imadx** [comment]
> HI, @imadx looks like this PR needs a rebase! Cheers, D

I've rebased and updated the PR. Looks okay now?

File: packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/calltoActionButton.stories.tsx
Comment (imadx) [reply packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/calltoActionButton.stories.tsx:1]: Updated this 🙌

File: packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.tsx
Comment (imadx) [reply packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.tsx:1]: Had to add those to avoid some cyclic reference issue.
Removed those unnecessary updates - doesn't seem to require them with the updated stories.

File: packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import styled from "@emotion/styled";
+
+export const Banner = styled.div`
+  background-color: ${({ theme }) => theme.palette.primary.main};
+  color: ${({ theme }) => theme.palette.common.white};
+  padding: 8px;
+  line-height: 1;
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:7]:
```suggestion
```

Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:7]: I think let's not try to control the line height from the parent component - the content should be driver for line height instead.

File: packages/data-explorer-ui/src/components/common/Button/button.styles.ts [hunk 10]
```diff
@@ -59,3 +59,13 @@ export const TextButtonPrimary = styled(MButtonBase)`
     text-decoration: underline;
   }
 `;
+
+// White text button
+export const TextButtonWhite = styled(MButtonBase)`
+  color: ${({ theme }) => theme.palette.common.white};
+  font-size: 13px;
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/button.styles.ts:66]:
```suggestion
```

File: packages/data-explorer-ui/src/components/common/Button/button.styles.ts [hunk 11]
```diff
@@ -59,3 +59,13 @@ export const TextButtonPrimary = styled(MButtonBase)`
...
+  font-weight: ${({ theme }) => theme.typography[textBody500].fontWeight};
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/button.styles.ts:67]:
```suggestion
```

File: packages/data-explorer-ui/src/components/common/Button/button.styles.ts [hunk 12]
```diff
@@ -59,3 +59,13 @@ export const TextButtonPrimary = styled(MButtonBase)`
...
+  line-height: ${({ theme }) => theme.typography[textBody500].lineHeight};
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/button.styles.ts:68]:
```suggestion
```

File: packages/data-explorer-ui/src/components/common/Button/button.styles.ts [hunk 13]
```diff
@@ -59,3 +59,13 @@ export const TextButtonPrimary = styled(MButtonBase)`
...
+  line-height: 1;
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/button.styles.ts:69]:
```suggestion
```

File: packages/data-explorer-ui/src/components/common/Button/button.styles.ts [hunk 14]
```diff
@@ -59,3 +59,13 @@ export const TextButtonPrimary = styled(MButtonBase)`
     text-decoration: underline;
   }
 `;
+
+// White text button
+export const TextButtonWhite = styled(MButtonBase)`
+  color: ${({ theme }) => theme.palette.common.white};
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/button.styles.ts:65]: insert above line `65` `${textBodySmall400};` and import from `import { textBodySmall400 } from "../../../styles/common/mixins/fonts";` these font styles will come in handy!

Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Button/button.styles.ts:65]: The `TextButtonPrimary` above was written prior to the availability of the fonts described above - we should try to use them as much as possible going forwards!

File: packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx [hunk 15]
```diff
@@ -1,6 +1,7 @@
-import { ComponentMeta, ComponentStory } from "@storybook/react";
+import { Meta, StoryFn } from "@storybook/react";
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx:1]:
```suggestion
import { Meta, StoryObj } from "@storybook/react";
```

File: packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx [hunk 16]
```diff
@@ -9,9 +10,9 @@ export default {
   },
   component: CallToActionButton,
   title: "Components/Common/CallToActionButton",
-} as ComponentMeta<typeof CallToActionButton>;
+} as Meta<typeof CallToActionButton>;
 
-const CallToActionButtonTemplate: ComponentStory<typeof CallToActionButton> = (
+const CallToActionButtonTemplate: StoryFn<typeof CallToActionButton> = (
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx:15]:
```suggestion
const CallToActionButtonTemplate: StoryObj <typeof CallToActionButton> = (
```

File: packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx [hunk 17]
```diff
@@ -31,6 +32,6 @@ CallToActionButtonStory.args = {
 export const CallToActionButtonWithCustomComponentStory =
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx:32]: Hi @imadx, let's leave `CallToActionButtonWithCustomComponentStory` as is, with the prop `ButtonElType` defined as `ButtonElType: "a"`. I think I will make another ticket to remove this story altogether, because we can override story arguments if we need to. If you could revert line `34` for me please, and I will create a new ticket to remove this story altogether.

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+const BannerTemplate: StoryFn<typeof Banner> = (args) => <Banner {...args} />;
+
+export const BannerStory = BannerTemplate.bind({});
+BannerStory.args = {
+  children: (
+    <>
+      <Typography component="span" variant="text-body-small-500">
+        Optional announcements text goes here followed by a
+      </Typography>{" "}
+      <CallToActionButtonWithCustomComponentStory
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:27]:
```suggestion
      <CallToActionButtonStory
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+        ButtonElType={
+          CallToActionButtonWithCustomComponentStory.args?.ButtonElType
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:29]:
```suggestion
          TextButtonWhiteStory
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 20]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        callToAction={
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:31]:
```suggestion
        callToAction={{
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+          CallToActionButtonWithCustomComponentStory.args?.callToAction || {
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:32]:
```suggestion
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+            label: "Call to Action",
+            url: "https://www.google.com",
+          }
+        }
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:36]:
```suggestion
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+      </Typography>{" "}
+      <CallToActionButtonWithCustomComponentStory
+        ButtonElType={
+          CallToActionButtonWithCustomComponentStory.args?.ButtonElType
+        }
+        callToAction={
+          CallToActionButtonWithCustomComponentStory.args?.callToAction || {
+            label: "Call to Action",
+            url: "https://www.google.com",
+          }
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:35]:
```suggestion
          }}
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+      />
+    </>
+  ),
+};
+
+export const BannerWithoutCallToActionTextStory = BannerTemplate.bind({});
+BannerWithoutCallToActionTextStory.args = {
+  children: (
+    <>
+      <Typography component="span" variant="text-body-small-500">
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:46]:
```suggestion
      <Typography component="span" variant="text-body-small-400">
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+  title: "Components/Common/Banner",
+} as Meta<typeof Banner>;
+
+const BannerTemplate: StoryFn<typeof Banner> = (args) => <Banner {...args} />;
+
+export const BannerStory = BannerTemplate.bind({});
+BannerStory.args = {
+  children: (
+    <>
+      <Typography component="span" variant="text-body-small-500">
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:24]:
```suggestion
      <Typography component="span" variant="text-body-small-400">
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:1]: Let's use the CallToActionStory like the following:

```
      <CallToActionButtonStory
        ButtonElType={TextButtonWhiteStory}
        callToAction={{
          label: "Call to Action",
          url: "https://www.google.com",
        }}
```

File: packages/data-explorer-ui/src/components/Layout/components/Header/components/ProfileComponent/components/AuthenticationMenu/authenticationMenu.stories.tsx (hunk 8 above)
Comment (frano-m) [reply packages/data-explorer-ui/src/components/Layout/components/Header/components/ProfileComponent/components/AuthenticationMenu/authenticationMenu.stories.tsx]: Hi @imadx, I'm not sure the code is reverted / rebased? I'm still seeing change requests for the story?

File: packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx (hunk 15 above)
Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx:1]: Would it be possible to revert the CallToActionButton change requests altogether?

File: packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx (hunk 17 above)
Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx:32]: Let's revert the packages/data-explorer-ui/src/components/common/Button/components/CallToActionButton/callToActionButton.stories.tsx altogether!

File: packages/data-explorer-ui/src/components/common/Banner/banner.tsx (hunk 5 above)
Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Banner/banner.tsx]: Hi @imadx, I think we should keep the Banner component as per the ticket [Part A](https://github.com/clevercanary/data-explorer/issues/89) - we may extend the use of the component - add props in the future etc etc. Can you please undo this last change?

File: packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts [hunk 26]
```diff
@@ -0,0 +1,9 @@
+import styled from "@emotion/styled";
+
+export const Banner = styled.div`
+  background-color: ${({ theme }) => theme.palette.primary.main};
+  color: ${({ theme }) => theme.palette.common.white};
+  padding: 8px;
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:6]: Insert between lines 5 and 6: `display: flex`

File: packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts [hunk 27]
```diff
@@ -0,0 +1,9 @@
...
+  line-height: 1;
+  text-align: center;
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:8]:
```suggestion
```

Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:8]: By removing the text alignment we give the children components complete control over how they wish to render themselves.

File: packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts (hunk 26 above)
Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:6]: And then, after the `display: flex` add `justify-content: center`.

Comment (frano-m) [reply packages/data-explorer-ui/src/components/common/Banner/banner.styles.ts:6]: There is a line-height on the `body` that the Banner is inheriting; the `display: flex` and alignment centre the content, and resolve the line-height problem (as well as using the correct Typography style for the content).

File: packages/data-explorer-ui/src/components/common/Button/button.styles.ts [hunk 28]
```diff
@@ -59,3 +59,13 @@ export const TextButtonPrimary = styled(MButtonBase)`
...
+  font-size: 13px;
+  font-weight: ${({ theme }) => theme.typography[textBody500].fontWeight};
+  line-height: ${({ theme }) => theme.typography[textBody500].lineHeight};
+  line-height: 1;
+  text-decoration: underline;
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Button/button.styles.ts:70]: Add the style `vertical-align: unset` - to override the unwanted alignment of the text from MuiButtonBase.

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+        }
+      />
+    </>
+  ),
+};
+
+export const BannerWithoutCallToActionTextStory = BannerTemplate.bind({});
+BannerWithoutCallToActionTextStory.args = {
+  children: (
+    <>
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:45]:
```suggestion
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 30]
```diff
@@ -0,0 +1,51 @@
...
+      <Typography component="span" variant="text-body-small-500">
+        Optional announcements text goes here
+      </Typography>{" "}
+    </>
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:49]:
```suggestion
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 31]
```diff
@@ -0,0 +1,51 @@
...
+  ),
+};
+
+export const BannerWithoutCallToActionTextStory = BannerTemplate.bind({});
+BannerWithoutCallToActionTextStory.args = {
+  children: (
+    <>
+      <Typography component="span" variant="text-body-small-500">
+        Optional announcements text goes here
+      </Typography>{" "}
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:48]:
```suggestion
      </Typography>
```

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 32]
```diff
@@ -0,0 +1,51 @@
...
+  },
+  title: "Components/Common/Banner",
+} as Meta<typeof Banner>;
+
+const BannerTemplate: StoryFn<typeof Banner> = (args) => <Banner {...args} />;
+
+export const BannerStory = BannerTemplate.bind({});
+BannerStory.args = {
+  children: (
+    <>
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:23]: Change the `children` argument to:

```
    <Typography component="span" variant="text-body-small-400">
      Optional announcements text goes here followed by a{" "}
      <CallToActionButtonStory
        ButtonElType={TextButtonWhiteStory}
        callToAction={{
          label: "Call to Action",
          url: "https://www.google.com",
        }}
      />
    </Typography>
```

I will make a separate ticket to make a new button component that inherits the parent typography styles.

**Review by frano-m — CHANGES_REQUESTED** [review]
Hi @imadx, there's quite a bit of feedback in the ticket:
1) I've made some suggestions that will resolve the line-height problem, including updating the typography to match the mocks
2) Let's keep the Banner component as per the original ticket requirements.
3) And then update the story as per feedback which should ultimately resolve all rendering problems.

Thanks so much 👍🏽

File: packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx [hunk 33]
```diff
@@ -0,0 +1,51 @@
+import { Typography } from "@mui/material";
+import { Meta, StoryFn } from "@storybook/react";
```
Comment (frano-m) [inline packages/data-explorer-ui/src/components/common/Banner/banner.stories.tsx:2]: Let's use `StoryObj` for the Banner story instead.

**Review by frano-m — APPROVED** [review]
